### ISSUE TITLE
Server errors when discord user has "." in nickname.

### ISSUE THREAD

**Issue by Krysinski** [shadowfacts/DiscordChat#71]
Brief explanation of issue: Server errors at console when discord user with "." in nickname send message from discord to minecraft, this message not send to minecraft.

Steps to reproduce:

- Step 1 Change your discord username to "User.test"
- Step 2 Send any message from discord chat.

Link to crash report/log in a [gist](https://gist.github.com): https://gist.github.com/Krysinski/b81f185b19c3d8b127acadb5b9b92dab

DiscordChat version: DiscordChat-1.7.10-2.0.3.jar

Minecraft version: Thermos 1.7.10-1614.56 (MC: 1.7.10)

**Comment by shadowfacts** [comment]
The cause of the error is Forge using a regex to match a URL to allow them to be clickable in chat. Forge doesn't have a whitelist of TLDs so anything following a pattern (such as your DC nickname, `Alirex.Prime`) will be thought to be a URL. Forge will then try to parse the URL properly which fails because it encounters the character `>` as part of the message format.

There is no complete solution to this, only a couple workarounds:

1. Stop parsing links. This allows nicks containing a period.
2. Continue parsing links. This prevents people from having nicks containing a period.

IMO, option 2 is better because having links be clickable in MC chat is better than not having them be clickable but allowing certain (rarely occurring) nicknames.

What do you think?

**Comment by Krysinski** [comment]
Here I completely agree with you. Could you add a warning about this problem in the description of your mod?

**Comment by hanakocz** [comment]
However you probably cannot guard all people on public servers to not use dot in name. On MC side it is easy, because MC does not allow that in names, but Discord... This probably should be done via bot, as he should eliminate the dot from parsed message author's nick to the server. Or transfer the dot to another harmless character.

**Comment by Krysinski** [comment]
I suppose that bot cant remove dot from nick's. But mod maybe can.

**Comment by Krysinski** [comment]
However mod parse messages from bot and get names. Then just need to replace "." with "_"
Something like this (bash):
```
 x=alirex.prime
 echo ${x//./_}
alirex_prime
```
or that
```
x=alirex.prime
x="${x//./_}"

echo $x
alirex_prime
```

**Comment by shadowfacts** [comment]
Fixed in DiscordChat 2.0.4
